app/models/line_item.rb

```diff
@@ -0,0 +1,54 @@
+class LineItem < ApplicationRecord
+  # Constants
+  MINIMUM_ALLOWED_QUANTITY = 1
+  # Associations
+  belongs_to :order, validate: true
+  belongs_to :deal
+  # Callbacks
+  after_rollback :add_errors_to_order
```

Comment:
didn't get the point, should not be here

Reply:
since i am performing update on order and never on line item, the failed validation of line item should be added to order's error messages.